nibabel/volumeutils.py

```diff
@@ -504,7 +507,8 @@ def array_from_file(shape, in_dtype, infile, offset=0, order='F', mmap=True):
             pass
     if len(shape) == 0:
         return np.array([])
-    n_bytes = int(np.prod(shape) * in_dtype.itemsize)
```

Comment:
The advantage of using `operator.mul` is that is uses Python integers internally, and so completely prevents overflow for any possible value of the shape.  In fact the test tests overflow for values above the int64 range, although of course you cannot make an array that large.